### DISCUSSION TITLE
Tracking Domains break email body if used with SMTP.

### DISCUSSION THREAD

**Discussion by uhlhosting** [postalserver/postal#1442] [Help with using Postal]
Hi,

We use a simple phpmailer script to send mails from our wordpress via Postal, some of our emails arrive like this;
```
Additional Headers

MIME-Version: 1.0
Content-type: text/html; charset=UTF-8
```

https://gist.github.com/uhlhosting/343e16da0ca6ae8abf89a3e8cb6b2273
I attached here the .eml broken one sent via postal

and here the normal one not sent via postal
https://gist.github.com/uhlhosting/4461e455e864e712c545d3468624f4ca

<img width="955" alt="image" src="https://user-images.githubusercontent.com/6324047/91584534-101b3f80-e953-11ea-9582-8aaf607974bf.png">

And are sent like:

<img width="747" alt="image" src="https://user-images.githubusercontent.com/6324047/91584564-1b6e6b00-e953-11ea-8ce3-aa80990ac4c0.png">

anyone has any ideas what could cause this?

**Comment by willpower232** [comment]
It sounds like there is some problem with character encoding which has been hinted at in some other issues. Do you have click tracking set up for these emails? Are you able to share the script you're using to send the messages?

**Comment by uhlhosting** [comment]
@willpower232 I have indeed click tracking enabled, i just use a phpmailer custom function in wordpress.

**Comment by willpower232** [comment]
We have experienced that sending email with SMTP into a click tracking server will cause problems outputting the email contents, can you confirm that disabling click tracking clears the problem?

**Comment by uhlhosting** [comment]
Yes disabling click solves it, just that is one case only, anything else works.

**Comment by dragoangel** [comment]
I has same issue here.

- 1:
```
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

0J/RgNC40LLQtdGCISDQndC10LrQvtGC0L7RgNGL0LUg0L/Qu9Cw0LPQuNC9
0Ysg0LHRi9C70Lgg0LDQstGC0L7QvNCw0YLQuNGH0LXRgdC60Lgg0L7QsdC9
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
000000000000000000000000000000000000000000000000000000000000
uSDRgdGC0L7RgNC+0L3RiyDQtNC10LnRgdGC0LLQuNC5INC90LUg0YLRgNC1
0LHRg9C10YLRgdGPLg0KDQoNCtCt0YLQuCDQv9C70LDQs9C40L3RiyDQvdC1
INGC0YDQtdCx0YPRjtGCINC+0LHQvdC+0LLQu9C10L3QuNGPOg0KLSBQYWdl
IEJ1aWxkZXIg0L7RgiBTaXRlT3JpZ2luINCy0LXRgNGB0LjRjyAyLjExLjMN
Cg0KDQrQldGB0LvQuCDQstGLINGB0YLQvtC70LrQvdGR0YLQtdGB0Ywg0YEg
0L/RgNC+0LHQu9C10LzQsNC80Lgg0Lgg0LLQsNC8INC/0L7QvdCw0LTQvtCx
0LjRgtGB0Y8g0L/QvtC00LTQtdGA0LbQutCwLCDQstCw0Lwg0L/QvtC80L7Q
s9GD0YIg0LLQvtC70L7QvdGC0ZHRgNGLINC90LAg0YTQvtGA0YPQvNCw0YUg
000000000000000000000000000000000000000000000000000000000000
a3Qva2luNUxFSFINCg0K0JrQvtC80LDQvdC00LAgV29yZFByZXNzDQoNCg==
```
note: part of base64 is changed to hide links, part of text still can be decoded.
It can decoded back if receiver can see that it base64 :D but this fail - 98% people will not get this.

- 2:
```
MIME-Version: 1.0
Content-Type: text/html; charset=UTF-8
Content-Transfer-Encoding: 8bit

<p><strong>From:</strong> <a href=3D"mailto:test@a.com">test</a> &#60;tes=
t@a.com&#62; </p>=0D
<p><strong>=D0=A1=D0=BE=D0=BE=D0=B1=D1=89=D0=B5=D0=BD=D0=B8=D0=B5:</stron=
g><br />=0D
this only test</p>=0D
=0D
<p class=3D'ampimg' style=3D'display:none;visibility:none;margin:0;paddin=
g:0;line-height:0;'><img src=3D'http://click.example.com/xx/yy' alt=3D''></p>=
```
Here text isn't correct. 

- If I disable TrackDomain, both text/plain and text/html will be send correctly:

```
<p><strong>From:</strong> <a href="mailto:test@a.com">test</a> &#60;test@a.com&#62; </p>
<p><strong>Сообщение:</strong><br />
this only test</p>
```
P.S.:
@uhlhosting when we know the case, maybe better change name of issue to something more correct: 
Tracking Domains break email body if used with SMTP.
@willpower232 can you please add bug label? Can this bug be pinned as it seriously can harm for people who doesn't know about it. Bad part that you said you already know about such issue, but due to postal has many issues open nobody can see serious ones. Also as no template was provided people write issues how they can without standard and labels isn't displays real issue types.

**Comment by dragoangel** [comment]
As I can see this related to `crlf` EOL symbol issue, it converted to `=0Dlf`.
Here what google says: https://askleo.com/why_does_my_email_sometimes_show_up_with_funny_characters_like_0d_in_it/

**Comment by dragoangel** [comment]
No update on issue? This blocks usage of track links 😢

**Comment by mattbaylor** [comment]
Just ran across this issue on another question. I'm commenting here to kind of bump this bug. It doesn't look like it's being tracked as an issue, which kinda indicates to me that solving this problem isn't much of a priority. Kinda bummed about that as this seems to be a really great tool, but this piece of functionality is critical to my use case...

https://github.com/postalserver/postal/discussions/1541

**Reply by uhlhosting** [reply]
Same here, same here.

**Comment by pjv** [comment]
Depending on your wordpress context, there may or may not be a work-around described in this issue: https://github.com/postalserver/postal/issues/171

For the sites I host on Wordpress that are configured to send via Postal (v1) with click-tracking enabled, the work-around described in that issue fixes the messed up html emails.

**Reply by uhlhosting** [reply]
Sorry, since 2017, so to keep using some other 3rd party plugins, in that case rather not. We are moving away from this to custom solutions built on K8s cluster. It seems it leaved us no choise. Thats the up and downs of FREE FOSS software, some people if it works for them, screw the rest, were the best...

**Comment by brywhi** [comment]
Following this thread. I'd also love to see this tracked as a true issue.

https://github.com/postalserver/postal/discussions/1783

**Comment by mattbaylor** [comment]
Cross posting this here... a little bit of movement WRT where we might adjust the code to account for this issue:
https://github.com/postalserver/postal/discussions/1541#discussioncomment-2151724

**Comment by The-3x** [comment]
experiencing the same issue. no solution yet? it should be marked as a critical bug.

**Comment by uhlhosting** [comment]
Hi @adamcooke no updates on this matter, we experience same from other mails sent via Postal with tracking enabled. Seems tracking works only on NON HTML Mail templates. Or it breaks the mail templates, when it updates the URL of the tracker.

**Comment by Trapulo** [comment]
I have same problem with emails sent by .NET Core System.Net.Mail.SmtpClient (Assembly System.Net.Mail, Version=6.0.0.0). This blocks using the system.. :(

is there any way to fix this?

**Reply by olegbliaher** [reply]
I had the same issue in PHPMailer and solved it by adding the following line:

`$mail->Encoding = 'quoted-printable';`

**Reply by dragoangel** [reply]
> I had the same issue in PHPMailer and solved it by adding the following line:
> 
> `$mail->Encoding = 'quoted-printable';`

The problem that you fixed that on sender side, not on server itself
